pkg/controller/catalog/storage/mem_storage_test.go

```diff
@@ -20,278 +20,242 @@ import (
 	"strings"
 	"testing"
 
-	model "github.com/kubernetes-incubator/service-catalog/model/service_controller"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	kapi "k8s.io/kubernetes/pkg/api"
 )
 
 const (
-	brokerOneUUID = "126b8154-a24a-4e79-9185-3df2eb4d18a8"
-	brokerTwoUUID = "2b0c42ed-c43a-4724-b883-e5ba878a8bfd"
+	brokerName1 = "Test1"
 )
 
 func TestNoBrokers(t *testing.T) {
-	s := CreateMemStorage()
-	l, err := s.ListBrokers()
+	const bogusBrokerName = "NOT THERE"
+	bs := CreateMemStorage().Brokers()
+	l, err := bs.List()
 	if err != nil {
-		t.Fatalf("ListBrokers failed with: %#v", err)
+		t.Fatalf("List failed with: %s", err)
 	}
 	if len(l) != 0 {
 		t.Fatalf("Expected 0 brokers, got %d", len(l))
 	}
-	b, err := s.GetBroker("NOT THERE")
+	b, err := bs.Get(bogusBrokerName)
 	if err == nil {
-		t.Fatal("GetBroker did not fail")
+		t.Fatal("Get did not fail")
 	}
 	if b != nil {
 		t.Fatalf("Got back a broker: %#v", b)
 	}
 }
 
 func TestAddBroker(t *testing.T) {
-	s := CreateMemStorage()
-	b := &model.ServiceBroker{GUID: "Test"}
-	cat := model.Catalog{
-		Services: []*model.Service{},
-	}
-	err := s.AddBroker(b, &cat)
+	bs := CreateMemStorage().Brokers()
+	b := &servicecatalog.Broker{ObjectMeta: kapi.ObjectMeta{Name: brokerName1}}
+	_, err := bs.Create(b)
 	if err != nil {
-		t.Fatalf("AddBroker failed with: %#v", err)
+		t.Fatalf("Create failed with: %s", err)
 	}
-	l, err := s.ListBrokers()
+	l, err := bs.List()
 	if len(l) != 1 {
 		t.Fatalf("Expected 1 broker, got %d", len(l))
 	}
-	b2, err := s.GetBroker("Test")
+	b2, err := bs.Get(brokerName1)
 	if err != nil {
-		t.Fatalf("GetBroker failed: %#v", err)
+		t.Fatalf("Get failed: %s", err)
 	}
 	if b2 == nil {
 		t.Fatal("Did not get back a broker")
 	}
-	if strings.Compare(b2.Name, b.Name) != 0 {
-		t.Fatalf("Names don't match, expected: '%s', got '%s'", b.Name, b2.Name)
+	if b2 != b {
```

Comment:
I wasn't saying to check one field, rather all fields - deep copy.

As long as this in-mem storage is just for fun/testing then I think its ok. But if we ever use it for anything more than I disagree. If I store a binding I don't think I should be able to modify it w/o going thru an Update() call. By just storing pointers we allow clients to modify resources in the storage w/o it knowing it - that's not good.